src/wiki/views/article.py

```diff
@@ -687,7 +688,17 @@ def dispatch(self, request, *args, **kwargs):
     def get_queryset(self):
         if not self.query:
             return models.Article.objects.none().order_by('-current_revision__created')
-        articles = models.Article.objects.filter(
+        articles = models.Article.objects
+        path = self.kwargs.get('path', None)
+        if path:
+            try:
+                self.urlpath = models.URLPath.get_by_path(path)
+                article_ids = self.urlpath.get_descendants(
+                    include_self=True).values_list('article_id')
+                articles = articles.filter(id__in=article_ids)
+            except (NoRootURL, models.URLPath.DoesNotExist):
```

Comment:
This should `raise Http404`, no?

Reply:
Yeah, it probably should. I'm not sure how to trigger that from a get_queryset method though.

Reply:
Oh that's fine, if you raise `Http404` anywhere in a view class, it'll get caught in the middleware. So to raise it here is okay.

Reply:
Oh, neat!